### DISCUSSION TITLE
Blocked atom is never mounted

### DISCUSSION THREAD

**Discussion by mwisnicki** [pmndrs/jotai#1802] [Q&A] [answered]
It looks like an atom will not mount until promise is resolved.

```ts
const anAtom = atom(
  new Promise<string>(() => {})
);
anAtom.onMount = (setAtom) => {
  console.log("mounted");
  setAtom(Promise.resolve("mounted"));
};
```

Above example will block forever. Is this expected?

https://codesandbox.io/s/jotai-forever-atom-never-mounted-9twwjh?file=/src/App.tsx

**Comment by dai-shi** [comment, marked as answer]
Yes, it's expected. While suspended, an atom will not be mounted. From react perspective, useEffect runs only after render, and it's only where we can "unmount" which should be paired with "mount".
